src/js/utils/DOM.js

```diff
@@ -31,6 +36,11 @@ export default {
     if (result.length === 0) {
       result.push(document);
     }
+    if (window.innerWidth <= Responsive.smallSize()
+      && result[0].classList
+      && result[0].classList.contains(LAYER_CLASS)) {
+      result = [document];
```

Comment:
@ericsoderberghp this checks if the user is viewing on mobile & the `scrollParent` is a Layer component. The problem is infinite scroll within the Layer component for mobile doesn't work because we need to set `scrollParent` to `document` instead of `.grommetux-layer` for scroll to work properly, since `.grommetux-layer` doesn't contain the scroll on mobile.  

In `InfiniteScroll.js` we're grabbing the first element from `result` which is `.grommetux-layer`, but this piece of code overrides `result` to be an array of just `document`, if the `scrollParent` is `.grommetux-layer`.